### PR TITLE
feat: add String::concat_view

### PR DESCRIPTION
**Pull request by Lampese**
- #1148

### CODE REVIEW

File: string/string.mbt [hunk 1]
```diff
@@ -64,6 +64,22 @@ pub fn String::concat(
   buf.to_string()
 }
```
Comment (bobzhang) [inline string/string.mbt:65]: what's the rationale to make it a plain function istead of a method?

**Comment by Lampese** [comment]
> How about:
> 
> ```
> pub(open) trait IterString {
>   iter(Self) -> Iter[String]
> }
> 
> pub fn concat[T : IterString](strings : T, separator~ : String = "") -> String {
>   let buf = StringBuilder::new()
>   let mut str : String? = None
>   for s in strings {
>     if str.is_empty().not() {
>       buf.write_string(str.unwrap())
>       buf.write_string(separator)
>     }
>     str = Some(s)
>   } else {
>     if str.is_empty().not() {
>       buf.write_string(str.unwrap())
>     }
>   }
>   buf.to_string()
> }
> 
> test {
>   inspect!(concat(([] : Array[String])), content="")
>   inspect!(concat(["a", "b", "c"]), content="abc")
>   inspect!(concat(["a", "b", "c"], separator="|"), content="a|b|c")
>   inspect!(concat(["a", "b", "c"][1:], separator="|"), content="b|c")
>   inspect!(
>     concat((["a", "b", "c"] : FixedArray[String]), separator="|"),
>     content="a|b|c",
>   )
> }
> ```

It is a quite feasible plan

**Comment by peter-jerry-ye** [comment]
We are now considering moving it to `Iter[String]::join`

**Comment by Yoorkin** [comment]
`String::concat` was a mistake. In the self parameter, the types in the type constructor can be a concrete type like String. After discussing offline, we will move to `Iter[String]::join` and make `String::concat` a function. The join method will be added later because users can always use other methods instead.

**Comment by Lampese** [comment]
- #1257

**Pull request closed without merging**